packages/visual-editor/src/components/puck/CollectionSection.tsx

```diff
@@ -0,0 +1,178 @@
+  };
+  sectionHeading: {
+    text: YextEntityField<string>;
+    level: HeadingProps["level"];
+  };
+  collection: {
+    items: YextEntityField<Array<any>>;
+    limit: string | number;
+  };
+}
+
+const collectionSectionFields: Fields<CollectionSectionProps> = {
+  styles: {
+    type: "object",
+    label: "Styles",
+    objectFields: {
+      background: BasicSelector(
+        "Background",
+        Object.values(backgroundColors).map(({ label, value }) => ({
+          label,
+          value,
+          color: value.bgColor,
+        }))
+      ),
+    },
+  },
+  sectionHeading: {
+    type: "object",
+    label: "Section Heading",
+    objectFields: {
+      text: YextEntityFieldSelector<any, string>({
+        label: "Section Heading Text",
+        filter: { types: ["type.string"] },
+      }),
+      level: BasicSelector("Heading Level", ThemeOptions.HEADING_LEVEL),
+    },
+  },
+  collection: {
+    type: "object",
+    label: "Collection",
+    objectFields: {
+      items: YextEntityFieldSelector<any, Array<any>>({
+        label: "Items",
+        isCollection: true,
+        filter: {
+          includeListsOnly: true,
+          // Filter hours fields because they have a lot of unused children
+          disallowList: [
+            "hours",
+            "accessHours",
+            "brunchHours",
+            "deliveryHours",
+            "dineInHours",
+            "driveThroughHours",
+            "happyHours",
+            "kitchenHours",
+            "onlineServiceHours",
+            "pickupHours",
+            "seniorHours",
+            "takeoutHours",
+          ],
+        },
+      }),
+      limit: OptionalNumberField({
+        fieldLabel: "Items Limit",
+        hideNumberFieldRadioLabel: "All",
+        hideNumberFieldRadioValue: "all",
+        showNumberFieldRadioLabel: "Limit",
+        defaultCustomValue: 3,
+      }),
+    },
+  },
+};
+
+const CollectionSectionWrapper: React.FC<CollectionSectionProps> = (props) => {
+  const { styles, sectionHeading, collection } = props;
+  const document = useDocument();
+
+  const resolvedHeadingText = resolveYextEntityField<string>(
+    document,
+    sectionHeading.text
+  );
+
+  return (
+    <Section background={styles.background}>
+      {resolvedHeadingText && (
+        <Heading level={sectionHeading.level}>{resolvedHeadingText}</Heading>
+      )}
+      <CollectionContext.Provider
+        value={{
+          parentEntityField: collection.items,
+          limit:
+            typeof collection.limit === "number" ? collection.limit : undefined,
+        }}
+      >
+        <DropZone
+          zone="collection-dropzone"
+          allow={["ExampleRepeatableItemComponent"]}
+          className="flex gap-4 max-w-pageSection-maxWidth flex-wrap"
+        />
+      </CollectionContext.Provider>
+    </Section>
+  );
+};
+
+export const CollectionSection: ComponentConfig<CollectionSectionProps> = {
+  label: "Collection Section",
```

Comment:
I wonder if this should really be a Section itself or it should really just be a wrapper of items and you'd put the Collection in a Section. I feel like the latter would be more flexible. That's a Product decision though.